lms/djangoapps/grades/transformer.py

```diff
@@ -49,6 +50,12 @@ def collect(cls, block_structure):
         """
         block_structure.request_xblock_fields(*cls.FIELDS_TO_COLLECT)
         cls._collect_max_scores(block_structure)
+        collect_unioned_set_field(
+            block_structure=block_structure,
+            transformer=cls,
+            merged_field_name='containing_subsections',
+            base_case_predicate=lambda block_key: block_key.block_type == 'sequential',
```

Comment:
`collect_unioned_set_field` is written to be general enough that this lambda is all you;d need to change in order to aggregate scores by vertical instead of subsection.